src/transformers/models/dpt/modeling_dpt.py

```diff
@@ -0,0 +1,1127 @@
+        pooled_output = self.pooler(sequence_output) if self.pooler is not None else None
+
+        if not return_dict:
+            return (sequence_output, pooled_output) + encoder_outputs[1:]
+
+        return BaseModelOutputWithPooling(
+            last_hidden_state=sequence_output,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+
+class DPTViTPooler(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.activation = nn.Tanh()
+
+    def forward(self, hidden_states):
+        # We "pool" the model by simply taking the hidden state corresponding
+        # to the first token.
+        first_token_tensor = hidden_states[:, 0]
+        pooled_output = self.dense(first_token_tensor)
+        pooled_output = self.activation(pooled_output)
+        return pooled_output
+
+
+class DPTNeck(nn.Module):
+    """
+    DPTNeck. A neck is a module that isused between the backbone and the head. It takes a list of tensors as input and
+    produces another list of tensors as output. For DPT, it includes:
+
+    * DPTReassembleBlocks
+    * FeatureFusionBlocks.
+
+    Args:
+        config (dict): config dict.
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.config = config
+
+        # postprocessing
+        self.reassemble_blocks = DPTReassembleBlocks(config)
+        self.post_process_channels = [
+            channel * math.pow(2, i) if config.expand_channels else channel
+            for i, channel in enumerate(config.post_process_channels)
+        ]
+        self.convs = nn.ModuleList()
+        for channel in self.post_process_channels:
+            self.convs.append(nn.Conv2d(channel, config.channels, kernel_size=3, padding=1, bias=False))
+
+        # fusion
+        self.fusion_blocks = nn.ModuleList()
+        for _ in range(len(self.convs)):
+            self.fusion_blocks.append(DPTFeatureFusionBlock(config))
+        self.fusion_blocks[0].res_conv_unit1 = None  # not sure why this is done in mmseg
+
+    def forward(self, hidden_states: List[torch.Tensor]) -> List[torch.Tensor]:
+        if len(hidden_states) != len(self.post_process_channels):
+            raise ValueError("The number of hidden states should be equal to the number of post-process channels.")
+
+        # postprocess hidden states
+        features = self.reassemble_blocks(hidden_states)
+
+        features = [self.convs[i](feature) for i, feature in enumerate(features)]
+
+        # fusion blocks
+        output = []
+        for i in range(len(self.fusion_blocks)):
+            if i == 0:
+                out = self.fusion_blocks[i](features[-1])
+            else:
+                out = self.fusion_blocks[i](out, features[-(i + 1)])
+            output.append(out)
```

Comment:
This can be simplified. Just start you loop one later and put line `796` before it